### DISCUSSION TITLE
Translate the volume to origin

### DISCUSSION THREAD

**Discussion by gungui98** [pyvista/pyvista#6016] [Q&A] [answered]
### First check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the PyVista documentation, with the integrated search.
- [X] I already searched in Google "How to X in PyVista" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I will NOT make duplicate discussions or issues in the future.

### Commit to Help

- [X] I commit to help with one of those options.

### Sample Code What is the problem, question, or error?
Write a short description telling me what you are doing, what you expect to happen, and what is currently happening.

```python
import numpy as np
import pyvista as pv
from pyvista import examples


# Download a volumetric dataset
vol = examples.download_knee_full()
print(vol.center)
print(vol.origin)
vol = vol.translate(-np.array(vol.center).astype(np.float64))

cpos = [(-381.74, -46.02, 216.54), (74.8305, 89.2905, 100.0), (0.23, 0.072, 0.97)]
pl = pv.Plotter()
pl.add_volume(vol, cmap="bone", opacity="sigmoid")
pl.add_point_labels(vol.center, ['Center'], point_color='blue', point_size=20, text_color='black', font_size=20,)
pl.add_point_labels(vol.origin, ['Origin'], point_color='red', point_size=20, text_color='black', font_size=20,)
pl.camera_position = cpos
pl.show()
```

### Description

I want to translate the volume that moves the center to the origin using the code, but when I visualize the volume, it is still in the same position. I also tried to assign the return value of `vol.translate` to the `vol` but the code seemed to freeze when rendering. 

### System Information

```shell
--------------------------------------------------------------------------------
  Date: Thu May 02 12:04:26 2024 SE Asia Standard Time
                OS : Windows
            CPU(s) : 12
           Machine : AMD64
      Architecture : 64bit
       Environment : IPython
        GPU Vendor : ATI Technologies Inc.
      GPU Renderer : AMD Radeon(TM) Graphics
       GPU Version : 4.5.14802 Core Profile Context 21.40.52 30.0.14052.5
  MathText Support : False
  Python 3.9.18 (main, Sep 11 2023, 14:09:26) [MSC v.1916 64 bit (AMD64)]
           pyvista : 0.43.6
               vtk : 9.3.0
             numpy : 1.26.4
        matplotlib : 3.8.2
            scooby : 0.9.2
             pooch : 1.8.1
            pillow : 9.4.0
           imageio : 2.34.1
           IPython : 8.18.1
             scipy : 1.11.4
      nest_asyncio : 1.5.8
--------------------------------------------------------------------------------
```

**Comment by tkoyama010** [comment, marked as answer]
Sorry for the inconvenience, the translate method does not seem to work well with the ImageData class. We recommend updating the origin property as follows instead.
```python
vol.origin = vol.origin - np.array(vol.center)
```

You can expect to get the results you request by executing the following code.

```python
import numpy as np
import pyvista as pv
from pyvista import examples


# Download a volumetric dataset
vol = examples.download_knee_full()
center = vol.center
origin = vol.origin
print(center)
print(origin)

# vol = vol.translate(-np.array(vol.center).astype(np.float64))
vol.origin = vol.origin - np.array(vol.center)

cpos = [(-381.74, -46.02, 216.54), (74.8305, 89.2905, 100.0), (0.23, 0.072, 0.97)]
pl = pv.Plotter()
pl.add_volume(vol, cmap="bone", opacity="sigmoid")
pl.add_point_labels(center, ['Center'], point_color='blue', point_size=20, text_color='black', font_size=20,)
pl.add_point_labels(origin, ['Origin'], point_color='red', point_size=20, text_color='black', font_size=20,)
pl.camera_position = cpos
pl.show()
```

**Reply by gungui98** [reply]
@tkoyama010 Thanks for your quick response, it seems to solve my problem. Another question, which may be unrelated to the post, but I am trying to visualize some markup points exported from the 3D slicer application to the pyvista, but the coordinate from keypoint doesn't seem to match with the coordinate of pyvista, I am using `pv.wrap(numpy_volume)` to load the volume data from dicom files, and the coordinate from 3D slicer is a tuple of R,A,S coordinates. Do you have any idea how to sync between the two applications?

**Reply by tkoyama010** [reply]
I'm sorry. I can't advise you on the 3D slicer; if you share the file you are trying to visualize in PyVista, I can advise you. A new thread would be better for getting advice from other community members.

**Comment by user27182** [comment]
> I am trying to visualize some markup points exported from the 3D slicer application to the pyvista, but the coordinate from keypoint doesn't seem to match with the coordinate of pyvista, I am using `pv.wrap(numpy_volume)` to load the volume data from dicom files, and the coordinate from 3D slicer is a tuple of R,A,S coordinates. Do you have any idea how to sync between the two applications?

In general medical images, such as DICOM images, have `spacing`, `origin`, and `direction` properties to correctly orient the image in physical xyz space. If you only load the image as a numpy array, you lose all of the other information, and the image will not be correctly defined.

When using 3D Slicer, it will correctly load of the orientation metadata from the DICOM image and transform it to standardized coordinates (e.g. RAS coordinates). Using `pv.wrap(numpy_volume)` does not do this.

To get your image coordinates in `3DSlicer` to match the coordinates in `pyvista` is not trivial. First, you should use `pv.read` directly with the DICOM instead of wrapping a numpy array, as this should correctly load the image properties. But, it still may not correctly load the image direction (see https://github.com/pyvista/pyvista/issues/5071).
